src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolderDescriptor.java

```diff
@@ -41,6 +41,40 @@ public String getDisplayName() {
     }
 
     /**
+     * Needed if it wants Folders are categorized in Jenkins 2.x.
+     *
+     * TODO: Override when the baseline is upgraded to 2.x
+     * TODO: Replace to {@code NestedProjectsCategory.ID}
```

Comment:
You mean, why not add the category to the plugin now? You could, if you wanted to make this plugin now depend on `2.0-beta-2` or similar, and switched its own version to a beta number. We would also need to make sure that the update center is set up for plugins with a 2.x core dep. In that case there is no need to put the category into core. But I see no particular advantage, since the main reason to do this would be to export a plugin API for the category ID, which means making _other_ plugins depend on the beta release of `cloudbees-folder` and thus core, so they in turn would need new beta releases. Probably too much work for now.